clamp/clamp.tex

```diff
@@ -0,0 +1,331 @@
+
+%Next we show that the stability relation as defined in Equation~\eqref{eq:relation} yields a valid transformation. %\grace{Oh I think I understand now we assume $d_{in} \leq d_{out}$}
+
+%\begin{lemma}[Soundness of the stability relation]\label{lemma:stability}
+% Give it a name
+%For every $(\din, \dout)$ pair, the function $\Relation(\din, \dout)$ returns $\False$ whenever there exists some $x$ such that $d_{\X}(x, x') \leq \din$ and $d_{\Y}(T(x), T(x')) > \dout$,
+%where $T$ denotes the clamping transformation. %and $\Relation(\din, \dout)$ the clamping stability relation. % defined in pseudocode xx; specify where the Relation is defined better
+%\end{lemma}
+
+% Must show that the relation fails for every $(d_{in}, d_{out})$ pair where there exists some $x$ such that $d_{\mathcal{X}}(x, x') \leq d_{in}$ and $d_{\mathcal{Y}}(T(x), T(x')) > d_{out}$.
+
+%\begin{proof}
+%Since clamping is 1-stable, $d_{\Y}(T(x), T(x')) \leq d_{\X}(x, x')$ for all $x, x' \in \X$. (This is a recursive argument with Theorem 2! I think it has to go)
+%\end{proof}
+        
+% MS EDIT1: I adjusted the statement to negate it-- the relation must fail for every pair that does not uphold the worst-case change in distances. It is fine for the relation to also fail for valid pairs-- there are many proofs out there that are not tight. Put another way, the relation need only fail everything that is not DP, it does not need to accept everything that is DP.
+
+% Should "input parameters" be more specified?
+% "appropriately quantified" sounds vague
+%\grace{Why is this called privacy guarantee and not stability something? I thought it should only be a "privacy guarantee" for measurements not transformations.}
+%\silvia{So ``privacy relation" for measurements $=$ ``stability relation" for transformations, but both are providing the DP guarantee. But maybe ``DP guarantee" is not the best name}
+
+\smallskip
+\textbf{(Domain-metric compatibility).}  For \texttt{MakeClamp}, both cases correspond to showing that \texttt{VectorDomain(T)} is compatible with symmetric distance. This follows directly from the definition of symmetric distance, as stated in \href{https://www.overleaf.com/project/60d214e390b337703d200982}{``List of definitions used in the proofs"}. The \textit{appropriate output domain property} shown above ensures that \texttt{output\_domain} is \texttt{VectorDomain(T)}, and hence also compatible with the \texttt{output\_metric} symmetric distance.
```

Comment:
I don't think you need to refer to the appropriate output domain property, which involves the function.  Here you just need to refer to the line that defines "output_domain = ..." (currently in preconditions) and check that it is compatible with the line "output_metric = ..."

Reply:
Ah right, I was thinking about this as showing compatibility between output_metric and the domain of function(v) (which should be the same as output_domain by 1.). But we are showing compatibility with output_domain, which is directly defined in the pseudocode.